bip-musig2.mediawiki

```diff
@@ -416,17 +416,17 @@ Input:
 * Run ''PartialSigVerifyInternal(psig, pubnonce<sub>i</sub>, pk<sub>i</sub>, session_ctx)''
 * Return success iff no failure occurred before reaching this point.
 
-'''''PartialSigVerifyInternal(psig, pubnonce, pk<sup>*</sup>, session_ctx)''''':
+'''''PartialSigVerifyInternal(psig, pubnonce, pk, session_ctx)''''':
 * Let ''(Q, gacc, _, b, R, e) = GetSessionValues(session_ctx)''; fail if that fails
 * Let ''s = int(psig)''; fail if ''s &ge; n''
-* Let ''R<sup>*</sup><sub>1</sub> = cpoint(pubnonce[0:33]), R<sup>*</sup><sub>2</sub> = cpoint(pubnonce[33:66])''
-* Let ''R<sup>*</sup>' = R<sup>*</sup><sub>1</sub> + b⋅R<sup>*</sup><sub>2</sub>''
-* Let ''R<sup>*</sup> = R<sup>*</sup>' '' if ''has_even_y(R)'', otherwise let ''R<sup>*</sup> = -R<sup>*</sup>' ''
-* Let ''P = cpoint(pk<sup>*</sup>)''; fail if that fails
+* Let ''R<sub>⁎,1</sub> = cpoint(pubnonce[0:33]), R<sub>⁎,2</sub> = cpoint(pubnonce[33:66])''
+* Let ''Re<sub>⁎</sub>' = R<sub>⁎,1</sub> + b⋅R<sub>⁎,2</sub>''
```

Comment:
merely tangentially related, but what's the rationale for calculating the LC with `b` in `PartialSigVerifyInternal`, as opposed to incorporating `b` into the partial sig's `pubnonce` output in `Sign()`?

Reply:
Sorry, I don't think I understand the question. Can you elaborate? What's a LC? What's the `pubnonce` output in `Sign()`? `Sign()`'s output is `psig`.

Reply:
eh, sorry, I meant for the partial signature verify internal component, as only the `s` value is actually returned. So what I was talking about is that for the nonce `R`, we use a linear combination (the aforementioned LC) of R' and R''. And I was merely saying that for `s`, we calculate the secret nonce summand with the linear combination taken into account, but for the partial signature verification, `R` is recalculated from the `pubnonce`, as opposed to simply getting passed in. My question was why recalculate it in that method, but that probably should have been a separate issue as opposed to a comment here.

Reply:
We can't trust the signer to compute `R` correctly. So if we took a malicious `R` from the output of `Sign()`, it could be that the corresponding `s` is correct for that `R` but that `R` doesn't match the `R` of the other signers, and then the partial sigs won't add up to a full signature.

What's possible is that if you're one signer in a session, and you have computed `R` during signing, then you can cache this value and reuse it when partially verifying another signer's partial sig.

Does this make sense?

Reply:
Makes perfect sense, thank you! I appreciate you taking the time to type it out here :)